### ISSUE TITLE
Extend tx data obtained from `/block` endpoint

### ISSUE THREAD

**Issue by marekrjpolak** [trezor/blockbook#818]
Hello,

in current CoinJoin WIP on Suite, we're fetching transactions from `/block` endpoint (for privacy reasons). Unfortunately, there are some data missing so we have to fetch all relevant transactions one by one again from `/tx` endpoint, which is a) inefficient and b) potentially privacy-leaking.

What would be ideal for us is to return the transactions from `/block` endpoint in the exact same format as from `/tx` endpoint (e.g. with some query param), or at least to be able to somehow deduce
- fee rate (`tx.size` and/or `tx.vsize` and/or `tx.hex` needed), and
- unspent outputs (currently we use `vin.txid` and `vin.vout` or `vin.coinbase` and analyze it ourselves)

I understand that it's not an easy thing to do (https://github.com/trezor/blockbook/issues/479), but still, is it feasible? Or is there any alternative how to obtain required data without querying specific addresses/xpubs/txids?

**Comment by martinboehm** [comment]
Hi @marekrjpolak, @matejkriz, what is your final decision about this request? What changes do you need? I will try to summarize what I remember from the discussion with @marekrjpolak and is feasible to do:

- for transactions returned from `/block` endpoint return `vin.txid`, `vin.vout` or `vin.coinbase`, `tx.size` and/or `tx.vsize`
- extend the websocket interface by `/block` endpoint equivalent (you should not be using the REST API)
- enable the websocket subscription for all transactions added to mempool

Is there something else I forgot?

**Comment by marekrjpolak** [comment]
Hi @martinboehm, thanks for bringing this up! I'll still have to think about it a bit, but you're mostly correct and didn't forget anything.

The only difference after today's talk with @szymonlesisz is that we may be able to calculate size/vsize from the transaction data we already have (e.g. we don't have neither hex nor scripts, but we have all the inputs' and outputs' addresses and therefore types which could theoretically be enough, no?). But I'd like to implement proof-of-concept first before I definitely say we don't need it on backend.

**Comment by marekrjpolak** [comment]
Hello again @martinboehm!

Still not finished, but it seems that we're able to calculate vsize from tx addresses well enough, so only the outpoints necessarily have to be added to the `/block` response.

**Comment by martinboehm** [comment]
@marekrjpolak Do you need the `vin.coinbase` data for coinbase transactions or it could be skipped?

**Comment by marekrjpolak** [comment]
> @marekrjpolak Do you need the `vin.coinbase` data for coinbase transactions or it could be skipped?

@martinboehm No, definitely not the data. Maybe just the fact that it's a coinbase input, but I can presume that every first (and only) input of the first transaction in a block is coinbase input, correct?

**Comment by martinboehm** [comment]
Yes, every first and only tx in a block is the coinbase tx. In addition to that, it will not have `vin.txid`.

**Comment by martinboehm** [comment]
Implemented by `-extendedindex` option.
